server/lib/nl/utils.py

```diff
@@ -650,6 +650,18 @@ def place_detection_with_heuristics(query_fn, query: str) -> List[str]:
         # heuristics above, for example.
         if "the " in p:
           p = p.replace("the ", "")
+
+        # If the detected place string needs to be replaced with shorter text,
+        # then do that here.
+        if p.lower() in constants.SHORTEN_PLACE_DETECTION_STRING:
+          p = constants.SHORTEN_PLACE_DETECTION_STRING[p.lower()]
+
+        # Also remove place text detected which is lexactly equal to some place types
+        # e.g. "states" etc. This is a shortcoming of place entity recognitiion libraries.
```

Comment:
Please include an example misbehavior this addresses?

Reply:
Provided an example